data/lib/core/item.lua

```diff
@@ -30,530 +30,743 @@ function Item.isTile(self)
-	['ExtraDefense'] = {key = ITEM_ATTRIBUTE_EXTRADEFENSE},
-	['Attack'] = {key = ITEM_ATTRIBUTE_ATTACK},
-	['AttackSpeed'] = {key = ITEM_ATTRIBUTE_ATTACK_SPEED},
-	['HitChance'] = {key = ITEM_ATTRIBUTE_HITCHANCE},
-	['ShootRange'] = {key = ITEM_ATTRIBUTE_SHOOTRANGE},
-	['Armor'] = {key = ITEM_ATTRIBUTE_ARMOR},
-	['Duration'] = {key = ITEM_ATTRIBUTE_DURATION, cmp = function(v) return v > 0 end},
-	['Text'] = {key = ITEM_ATTRIBUTE_TEXT, cmp = function(v) return v ~= '' end},
-	['Date'] = {key = ITEM_ATTRIBUTE_DATE},
-	['Writer'] = {key = ITEM_ATTRIBUTE_WRITER, cmp = function(v) return v ~= '' end}
-}
-
-function setAuxFunctions()
-	for name, def in pairs(aux) do
-		Item['get'.. name] = function(self)
-			local attr = self:getAttribute(def.key)
-			if def.cmp and def.cmp(attr) then
-				return attr
-			elseif not def.cmp and attr and attr ~= 0 then
-				return attr
-			end
-			local default = ItemType['get'.. name]
-			return default and default(self:getType()) or nil
+	setmetatable(StringStream, {
+		__call = function(self)
+			local obj = {}
+			return setmetatable(obj, {__index = StringStream})
 		end
+	})
+
+	function StringStream.append(self, str, ...)
+		self[#self+1] = string.format(str, ...)
 	end
-end
 
-setAuxFunctions()
+	function StringStream.concat(self, sep)
+		return table.concat(self, sep)
+	end
 
-do
 	local function internalItemGetNameDescription(it, item, subType, addArticle)
-		local subType = subType or (item and item:getSubType() or -1)
+		subType = subType or (item and item:getSubType() or -1)
 		local ss = StringStream()
 		local obj = item or it
 		local name = obj:getName()
-		if name ~= '' then
+		if name ~= "" then
 			if it:isStackable() and subType >  1 then
 				if it:hasShowCount() then
-					ss:append('%d ', subType)
+					ss:append("%d ", subType)
 				end
-				ss:append('%s', obj:getPluralName())
+				ss:append("%s", obj:getPluralName())
 			else
-				if addArticle and obj:getArticle() ~= '' then
-					ss:append('%s ', obj:getArticle())
+				if addArticle and obj:getArticle() ~= "" then
+					ss:append("%s ", obj:getArticle())
 				end
-				ss:append('%s', obj:getName())
+				ss:append("%s", obj:getName())
 			end
 		else
-			ss:append('an item of type %d', obj:getId())
+			ss:append("an item of type %d", obj:getId())
 		end
 		return ss:concat()
 	end
 
-	function Item.getNameDescription(self, subType, addArticle)
+	function Item:getNameDescription(subType, addArticle)
 		return internalItemGetNameDescription(self:getType(), self, subType, addArticle)
 	end
 
-	function ItemType.getNameDescription(self, subType, addArticle)
+	function ItemType:getNameDescription(subType, addArticle)
 		return internalItemGetNameDescription(self, nil, subType, addArticle)
 	end
 end
 
 do
-	local function addSeparator(ss, begin)
-		if begin then
-			begin = false
-			ss:append(' (')
-		else
-			ss:append(', ')
+	local housePriceVisible = configManager.getBoolean(configKeys.HOUSE_DOOR_SHOW_PRICE)
```

Comment:
😡

Reply:
@Zbizu 
![image](https://user-images.githubusercontent.com/1552144/146914093-198b2710-fd81-4403-83dd-05a50d53e372.png)

I believe credits are propagating appropriately through git, especially when done correctly. With 144 contributors on this repo, it would be messy to have tons of comments everywhere telling who did what, especially since we got tools like git blame to figure it out.

Reply:
explain npc libs then